ops/preprocessing_smk.py

```diff
@@ -1,11 +1,12 @@
 import inspect
 import functools
 import os
-import warnings
+import re
+
 import pandas as pd
 import numpy as np
-import re
 from nd2reader import ND2Reader

```

Comment:
The one change we want to make here is switching from nd2Reader to nd2. The tile based pre-processing is way easier than the well based one, so you may want to weigh up if you want to make that switch when you rewrite in the new repo: https://www.talleylambert.com/nd2/API/nd2/. Talley Lambert is one of the most well known computational microscopy people around, and we can be sure that this package will be maintained long term

Reply:
Thanks for pointing this out! I like the idea of switching from nd2Reader to nd2 when rewriting preprocessing in the new repo.